### PR TITLE
Implement checkbox VPN exclusions UI

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/0/1206580121312550/1208221568417241/f

## Description

- Implements the checkbox-style UI for VPN site exclusions.
- When a domain exclusion is changed by the user we refresh the active tab.
- Some internal adjustments to fix an issue that was causing proxy configurations to be duplicated in 🍎 > System Settings... > Network > VPN & Filters > Filters & Proxies

## Design

We're updating the "Website prefrences" section shown in this image

Screenshot 2024-09-04 at 12 13 29 PM

## Testing

1. Before starting the VPN, open duckduckgo.com and search for "what is my ip" so you can see your IP within duckduckgo.com
2. Start the VPN
3. Reload the page showing your IP, and the IP should change to the VPN one
4. Go to the browser's navigation bar VPN menu, and tick the checkbox so the sire is excluded from the VPN
5. The site should reload on its own and show your public IP address
6. You can repeat the step where you exclude and include the site from the VPN to see it reloads and shows the right IP address
7. Go to 🍎 > System Settings... > Network > VPN & Filters > Filters & Proxies and ensure you only see one proxy configuration.  It may happen that you see more if you have used VPN exclusions in the past... remove them all manually and test again if that happens.

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/VPNUIActionHandler.swift [hunk 1]
```diff
@@ -49,6 +49,10 @@ final class VPNUIActionHandler: VPNUIActionHandling {
     func setExclusion(_ exclude: Bool, forDomain domain: String) async {
         proxySettings.setExclusion(exclude, forDomain: domain)
         try? await vpnIPCClient.command(.restartAdapter)
+
+        Task {
+            await vpnURLEventHandler.reloadTab(showingDomain: domain)
+        }
```
Comment (diegoreymendez) [inline DuckDuckGo/NetworkProtection/AppTargets/BothAppTargets/VPNUIActionHandler.swift:55]: Trigger a tab reload, but don't make the parent caller await on this.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionProxy/TransparentProxyController.swift [hunk 2]
```diff
@@ -168,37 +159,36 @@ public final class TransparentProxyController {
...
+            return manager
         }
     }
 
     /// Loads an existing configuration or creates a new one, if one doesn't exist.
     ///
     /// - Returns a properly configured `NETransparentProxyManager`.
     ///
-    public func loadOrCreateConfiguration() async throws -> NETransparentProxyManager {
+    public func loadOrMakeManager() async throws -> NETransparentProxyManager {
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionProxy/TransparentProxyController.swift:174]: Just a rename that helps readability.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionProxy/TransparentProxyController.swift [hunk 3]
```diff
@@ -168,37 +159,36 @@ public final class TransparentProxyController {
...
-                    (manager.protocolConfiguration as? NETunnelProviderProtocol)?.providerBundleIdentifier == extensionID
-                }
-                self.internalManager = manager
+            let manager = try? await NETransparentProxyManager.loadAllFromPreferences().first { manager in
+                (manager.protocolConfiguration as? NETunnelProviderProtocol)?.providerBundleIdentifier == extensionID
             }
-
-            return internalManager
+            internalManager = manager
+            return manager
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionProxy/TransparentProxyController.swift:166]: I'm changing the code in this file to be more similar to code from `NetworkProtectionTunnelController` which is doing the same things for the VPN tunnel.  We had a bug here that was causing the configuration to be duplicated, likely caused by the `triedLoadingManager` class.

File: LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/SiteTroubleshootingView/SiteTroubleshootingView.swift [hunk 4]
```diff
@@ -41,57 +38,61 @@ public struct SiteTroubleshootingView: View {
...
-                                    .frame(width: 16, height: 16)
-                                    .applyCurrentSiteAttributes()
-                            } else {
-                                Rectangle()
-                                    .fill(Color.clear)
-                                    .frame(width: 16, height: 16)
-                            }
-                            Text("Exclude from VPN")
-                        }
-                    }
```
Comment (diegoreymendez) [inline LocalPackages/NetworkProtectionMac/Sources/NetworkProtectionUI/Views/SiteTroubleshootingView/SiteTroubleshootingView.swift:72]: The accordion view is no more, we settled on the checkbox.

**Review by jotaemepereira — APPROVED** [review]
The UI changes look good 👍🏼 .

I found a bug that @diegoreymendez is taking care of here: [una task para los refresh issues](https://app.asana.com/0/0/1208074698306034/f)

given that the bug is being worked on another task, I’m going to approve this ✅

**Pull request merged**